src/library/scala/collection/immutable/Vector.scala

```diff
@@ -1146,7 +1158,7 @@ private[immutable] trait VectorPointer[+T] {
     * If `destination` array is not null, original contents of array(index) will be copied to it, and it will be returned.
     * Otherwise array(index).clone() is returned
     */
-    private[immutable] final def nullSlotAndCopy[T <: AnyRef](array: Array[Array[T]], index: Int, destination: Array[T] = null): Array[T] = {
+    private[immutable] final def nullSlotAndCopy[T0 <: AnyRef](array: Array[Array[T0]], index: Int, destination: Array[T0] = null): Array[T0] = {
```

Comment:
The change `T` -> `T0` is simply to avoid shadowing which is going on between this `T` and the `T` in `VectorPointer[T]`